### PR TITLE
add concept doc for rbac

### LINKED ISSUE #570: Add Documentation for Registry Access Control
Currently, only Readme.md in access_control folder explains the RBAC usage. 
Users need more guidance on 

- What is Feathr Registry Access Control
- Basic concepts 
- How to Enable

### PR DESCRIPTION
**Pull request by Yuqing-cat**
## Description

Resolves #570

## How was this patch tested?

This PR only introduce a new markdown doc.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

## Dependencies

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to list all the dependencies and licenses.

### CODE REVIEW

File: docs/concepts/registry-access-control.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+A project level role based access control (RBAC) plugin is available to help you manage who have access to **Feathr Registry**.
+
+It provides a simple authorization system built on OAuth tokens along with a SQL database as backend storage for `userrole` records.
+
+
+## How Registry Access Control works?
+
+With Registry Access Control, you can
+- Share your project metadata(read-only) by assigning `consumer` role.
+- Invite contributors to update features in this project by assigning `producer` role.
```
Comment (xiaoyongzhu) [inline docs/concepts/registry-access-control.md:18]: make it clear that `producer` will have write access?

File: docs/concepts/registry-access-control.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+- Transfer project admin by assigning `admin` role.
```
Comment (xiaoyongzhu) [inline docs/concepts/registry-access-control.md:19]: talk a bit on what permissions that `admin` has

File: docs/concepts/registry-access-control.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+### Scope
+
+_Scope_ is the set of resources that the permission applies to. In Feathr, you can specify a scope at 2 levels: **registry**, **project**. Lower levels inherit role permissions from higher levels.
```
Comment (xiaoyongzhu) [inline docs/concepts/registry-access-control.md:23]: what does the `registry` part mean here? I guess it's global?

File: docs/concepts/registry-access-control.md (hunk 2 above)
Comment (xiaoyongzhu) [reply docs/concepts/registry-access-control.md:19]: Oh I see there are explanations below. Might be good to add links here.

File: docs/concepts/registry-access-control.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+### Permission
+_permission_ refers to the a certain kind of access to registry metadata or role assignment records.
+| Permission | Description                                               |
+| ---------- | --------------------------------------------------------- |
+| Read       | Read registry meta data; `GET` Registry APIs              |
+| Write      | Write registry meta data; `POST` Registry APIs            |
+| Manage     | Create and manage role assignment records with management APIs |
+
+### User
+A _user_ can be an email account or an AppId.
```
Comment (xiaoyongzhu) [inline docs/concepts/registry-access-control.md:54]: Explain what is AppId

File: docs/concepts/registry-access-control.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+| ---------- | --------------------------------------------------------- |
+| Read       | Read registry meta data; `GET` Registry APIs              |
+| Write      | Write registry meta data; `POST` Registry APIs            |
+| Manage     | Create and manage role assignment records with management APIs |
+
+### User
+A _user_ can be an email account or an AppId. 
+
+All Registry API requests requires `token` in header to identify the requestor and validate the permission.
+Feathr UI uses the id token of login account. While Feathr Client samples get token by `DefaultAzureCredential()`.
```
Comment (xiaoyongzhu) [inline docs/concepts/registry-access-control.md:57]: What does this mean?

Comment (xiaoyongzhu) [reply docs/concepts/registry-access-control.md:57]: `While Feathr Client samples get token by `DefaultAzureCredential()`.` - in particular this part...

File: docs/concepts/registry-access-control.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
+---
+layout: default
+title: Registry Access Control
+parent: Feature Registry
```
Comment (xiaoyongzhu) [inline docs/concepts/registry-access-control.md:4]: maybe put it into the concept folder to make the hierarchy more flat.

File: docs/concepts/registry-access-control.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+| Write      | Write registry meta data; `POST` Registry APIs            |
+| Manage     | Create and manage role assignment records with management APIs |
+
+### User
+A _user_ can be an email account or an AppId. 
+
+All Registry API requests requires `token` in header to identify the requestor and validate the permission.
+Feathr UI uses the id token of login account. While Feathr Client samples get token by `DefaultAzureCredential()`.
+
+_Group_ is **NOT** supported yet.
```
Comment (xiaoyongzhu) [inline docs/concepts/registry-access-control.md:59]: Should we explicitly it's AAD group?

File: docs/concepts/registry-access-control.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+Feathr UI uses the id token of login account. While Feathr Client samples get token by `DefaultAzureCredential()`.
+
+_Group_ is **NOT** supported yet. 
+
+### Role Assignment
+
+A _Role Assignment_ is the process of add a `user-role` mapping record into backend storage table.
+
+[Feature Registry](https://linkedin.github.io/feathr/concepts/feature-registry.html#access-control-management-page) section briefly introduced the access control management page, where project admins can manage role assignments. 
+Management APIs are not exposed in Feathr Client by design.
```
Comment (xiaoyongzhu) [inline docs/concepts/registry-access-control.md:66]: Maybe touch a bit on the data plane and control plane?

File: docs/concepts/registry-access-control.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+
+## How to enable Registry Access Control?
+[Azure Resource Provisioning](https://linkedin.github.io/feathr/how-to-guides/azure-deployment-arm.html) section has detailed instructions on resource provisioning. For RBAC specific, you will:
+1. Choose `Yes` for `Enable RBAC` in ARM Template.
+2. Create a `userrole` table in provisioned SQL database with [RBAC Schema](../../registry/access_control/scripts/schema.sql).
```
Comment (xiaoyongzhu) [inline docs/concepts/registry-access-control.md:72]: say explicitly that users should do this manually?

**Review by xiaoyongzhu — APPROVED** [review]

**Review by blrchen — APPROVED** [review]

**Pull request merged**